operators/vtk_renderer/vtk_renderer.cpp

```diff
@@ -0,0 +1,189 @@
+    if (cudaSuccess != _holoscan_cuda_err) {                                             \
+      GXF_LOG_ERROR("CUDA Runtime call %s in line %d of file %s failed with '%s' (%d).", \
+                    #stmt,                                                               \
+                    __LINE__,                                                            \
+                    __FILE__,                                                            \
+                    cudaGetErrorString(_holoscan_cuda_err),                              \
+                    _holoscan_cuda_err);                                                 \
+    }                                                                                    \
+    _holoscan_cuda_err;                                                                  \
+  })
+
+/**
+ * @brief Render text at a specific location on the renderer.
+ *
+ * @param text The text to render.
+ * @param x The x coordinate of the text.
+ * @param y The y coordinate of the text.
+ * @param renderer The renderer to render the text on.
+ */
+static void render_text_at_location(std::string text, float x, float y, vtkRenderer* renderer) {
+  vtkNew<vtkTextActor> textActor;
+  textActor->SetInput(text.c_str());
+  textActor->GetPositionCoordinate()->SetCoordinateSystemToNormalizedViewport();
+  textActor->GetPositionCoordinate()->SetValue(x, y);
+  renderer->AddActor(textActor);
+}
+
+/**
+ * @brief Reposition the camera to fit the image in the renderer.
+ *
+ * @param renderer The renderer to reposition the camera in.
+ * @param imageData The image data to fit in the renderer.
+ */
+static void reposition_camera(vtkRenderer* renderer, vtkImageData* imageData) {
+  // Set up the background camera to fill the renderer with the image.
+  double origin[3];
+  double spacing[3];
+  int extent[6];
+  imageData->GetOrigin(origin);
+  imageData->GetSpacing(spacing);
+  imageData->GetExtent(extent);
+
+  vtkCamera* camera = renderer->GetActiveCamera();
+  camera->ParallelProjectionOn();
+
+  // Center the image in the renderer.
+  double xc = origin[0] + 0.5 * (extent[0] + extent[1]) * spacing[0];
+  double yc = origin[1] + 0.5 * (extent[2] + extent[3]) * spacing[1];
+  double yd = (extent[3] - extent[2] + 1) * spacing[1];
+  double d = camera->GetDistance();
+  camera->SetParallelScale(0.5 * yd);
+  camera->SetFocalPoint(xc, yc, 0.0);
+  camera->SetPosition(xc, yc, d);
+}
+
+namespace holoscan::ops {
+
+void VtkRendererOp::setup(OperatorSpec& spec) {
+  auto& in_annotations = spec.input<gxf::Entity>("annotations");
+  auto& in_videostream = spec.input<gxf::Entity>("videostream");
+
+  spec.param(this->in_annotations,
+             "in_annotations",
+             "Input Annotations",
+             "Input Annotations",
+             &in_annotations);
+  spec.param(this->in_videostream,
+             "in_videostream",
+             "Input Videostream",
+             "Input Videostream",
+             &in_videostream);
```

Comment:
The spec.param() here is not needed. I assume this is from the HolovizOp which has the `receivers` input which can be connected to multiple outputs. Since you split the inputs to two you don't need this.
```suggestion
  spec.input<gxf::Entity>("annotations");
  spec.input<gxf::Entity>("videostream");
```